lib/openstudio-standards/hvac_sizing/Siz.HeatingCoolingFuels.rb

```diff
@@ -196,6 +196,358 @@ def coil_heating_fuels(heating_coil)
+          heat_type = 'HeatPump'
+        elsif heat_type != 'HeatPump'
+          heat_type = 'Mixed'
+        end
+      end
+    end
+
+    return heat_type
+  end
+
+  # Get primary type of heat for one air loop for Appendix G system type selections
+  # 
+  # @param air_loop [object] 
+  # @return string description of heat type: Fuel, Electric, District, HeatPump, Solar, Mixed
+  def airloop_primary_heat_type(air_loop)
+
+    heat_types = []
+    air_loop.supplyComponents.each do |component|
+      # Get the object type
+      obj_type = component.iddObjectType.valueName.to_s
+      case obj_type
+      when 'OS_AirLoopHVAC_UnitaryHeatCool_VAVChangeoverBypass'
+        component = component.to_AirLoopHVACUnitaryHeatCoolVAVChangeoverBypass.get
+        heat_types << self.coil_heat_type(component.heatingCoil)
+      when 'OS_AirLoopHVAC_UnitaryHeatPump_AirToAir'
+        component = component.to_AirLoopHVACUnitaryHeatPumpAirToAir.get
+        heat_types << self.coil_heat_type(component.heatingCoil)
+      when 'OS_AirLoopHVAC_UnitaryHeatPump_AirToAir_MultiSpeed'
+        component = component.to_AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed.get
+        heat_types << self.coil_heat_type(component.heatingCoil)
+      when 'OS_AirLoopHVAC_UnitarySystem'
+        component = component.to_AirLoopHVACUnitarySystem.get
+        if component.heatingCoil.is_initialized
+          heat_types << self.coil_heat_type(component.heatingCoil.get)
+        end
+      when 'OS_Coil_Heating_DX_MultiSpeed'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_DX_SingleSpeed'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_DX_VariableSpeed'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Desuperheater'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Electric'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Gas'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Gas_MultiStage'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_Water'
+        heat_types << self.coil_heat_type(component)  
+      when 'OS_Coil_Heating_WaterToAirHeatPump_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_WaterToAirHeatPump_VariableSpeed_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_AirToWaterHeatPump'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_Desuperheater'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Node', 'OS_Fan_ConstantVolume', 'OS_Fan_VariableVolume', 'OS_AirLoopHVAC_OutdoorAirSystem'
+        # To avoid extraneous debug messages  
+      else
+        #OpenStudio::logFree(OpenStudio::Debug, 'openstudio.sizing.Model', "No heating fuel types found for #{obj_type}")
```

Comment:
add warning message here if no heat type found

Reply:
I think no heat type is a valid result. In that case the method returns an empty string.